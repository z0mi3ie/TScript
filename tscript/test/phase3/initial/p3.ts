//print = function (x) { console.log(x); };

//BasicObject = function () { return this; };
//var a;
//print a;

BasicObject = 42;

BattleOfHastingsPrototype = new BasicObject();
BattleOfHastingsPrototype.date = 1066;

//BattleOfHastings = function () { return this; };
BattleOfHastings = new BasicObject();
BattleOfHastings.prototype = BattleOfHastingsPrototype;

var z;
z = new BattleOfHastings();

print (z.date);

print (NaN);
print (Infinity);
print (undefined);

print (NaN + Infinity);

