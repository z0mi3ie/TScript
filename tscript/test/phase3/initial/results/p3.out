1066
NaN
Infinity
undefined
NaN
